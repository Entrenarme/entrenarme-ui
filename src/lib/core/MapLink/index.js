// @flow
import React from 'react';
import styled from 'styled-components';
import JssProvider from 'react-jss/lib/JssProvider';
import { create } from 'jss';
import {
  createGenerateClassName,
  jssPreset,
  createMuiTheme,
  MuiThemeProvider,
} from '@material-ui/core/styles';

import Button from '../Button/index';
import { RESOURCES_URL } from '../../helpers/config';

const generateClassName = createGenerateClassName({
  productionPrefix: 'maplink-eui',
});
const jss = create(jssPreset());

const theme = createMuiTheme();

const MainContainer = styled.div`
  position: relative;
  width: 100%;

  img {
    width: 100%;
    height: auto;
  }
`;

const ExtendedImage = styled.img`
  width: 100%;
`;

const ExtendedButton = styled(Button)`
  position: absolute !important;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%) !important;
  -ms-transform: translate(-50%, -50%) !important;
`;

type Props = {
  buttonTitle: string,
  onClick: Function,
};

const MapLink = ({ buttonTitle, onClick }: Props) => {
  return (
    <JssProvider jss={jss} generateClassName={generateClassName}>
      <MuiThemeProvider theme={theme}>
        <MainContainer>
          <ExtendedImage
            alt="map"
            src={`${`${RESOURCES_URL}static/images/list/map.svg`}`}
          />
          <ExtendedButton type="outline" onClick={() => onClick()}>
            {buttonTitle}
          </ExtendedButton>
        </MainContainer>
      </MuiThemeProvider>
    </JssProvider>
  );
};

MapLink.defaultProps = {};

export default MapLink;
