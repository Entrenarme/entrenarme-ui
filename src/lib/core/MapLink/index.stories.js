// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import MapLink from './index';

storiesOf('MapLink', module).add(
  'Default',
  withInfo('MapLink')(() => (
    <MapLink buttonTitle="Ver mapa" onClick={() => console.log('hola!')} />
  )),
);
