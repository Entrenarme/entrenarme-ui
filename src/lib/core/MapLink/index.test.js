import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import MapLink from './index';

test('renders "MapLink" with buttonTitle', () => {
  const { container } = render(<MapLink buttonTitle="Ver mapa" />);

  expect(container.firstChild).toMatchSnapshot();
});
