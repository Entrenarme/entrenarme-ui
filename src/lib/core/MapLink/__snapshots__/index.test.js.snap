// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "MapLink" with buttonTitle 1`] = `
<div
  class="sc-htpNat jOIFTF"
>
  <img
    class="sc-bxivhb jcNPuI"
    alt="map"
    src="https://resources.entrenar.me/static/images/list/map.svg"
  />
  <button
    tabindex="0"
    class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-ifAKCX hSbmtv sc-bdVaJa gyItIq"
    type="button"
    options="[object Object]"
  >
    <span
      class="MuiButton-label-2"
    >
      <span>
        Ver mapa
      </span>
    </span>
    <span
      class="MuiTouchRipple-root-28"
    />
  </button>
</div>
`;
